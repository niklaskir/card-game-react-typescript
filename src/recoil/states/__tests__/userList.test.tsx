import React from 'react';
import renderer from 'react-test-renderer';
import { RecoilRoot, useRecoilValue, useSetRecoilState } from 'recoil';
import {
  sortedUserList as sortedUserListState,
  userListSortType as userListSortTypeState,
} from '../userList';

const DefaultUserList = () => {
  const sortedUserList = useRecoilValue(sortedUserListState);

  return (
    <>
      {sortedUserList.map((user, index) => (
        <p key={index}>{user.realName}</p>
      ))}
    </>
  );
};

const AscUserList = () => {
  const sortedUserList = useRecoilValue(sortedUserListState);
  const setUserSort = useSetRecoilState(userListSortTypeState);
  setUserSort('Asc');

  return (
    <>
      {sortedUserList.map((user, index) => (
        <p key={index}>{user.realName}</p>
      ))}
    </>
  );
};

const DescUserList = () => {
  const sortedUserList = useRecoilValue(sortedUserListState);
  const setUserSort = useSetRecoilState(userListSortTypeState);
  setUserSort('Desc');

  return (
    <>
      {sortedUserList.map((user, index) => (
        <p key={index}>{user.realName}</p>
      ))}
    </>
  );
};

describe('userList', () => {
  it('should render default user list', () => {
    const defaultUserList = renderer
      .create(
        <RecoilRoot>
          <DefaultUserList />
        </RecoilRoot>,
      )
      .toJSON();

    expect(defaultUserList).toMatchSnapshot();
  });

  it('should render asc user list', () => {
    const ascUserList = renderer.create(
      <RecoilRoot>
        <AscUserList />
      </RecoilRoot>,
    );

    expect(ascUserList).toMatchSnapshot();
  });

  it('should render desc user list', () => {
    const descUserList = renderer
      .create(
        <RecoilRoot>
          <DescUserList />
        </RecoilRoot>,
      )
      .toJSON();

    expect(descUserList).toMatchSnapshot();
  });
});
