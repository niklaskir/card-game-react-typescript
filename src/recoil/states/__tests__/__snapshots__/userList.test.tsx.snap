// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userList should render asc user list 1`] = `
Array [
  <p>
    Brianna Forbes
  </p>,
  <p>
    Darcy Candice Ball
  </p>,
  <p>
    Elva Becky Hammond
  </p>,
  <p>
    Enid Rose
  </p>,
  <p>
    Esmeralda Carrillo
  </p>,
  <p>
    Hillary Gibbs
  </p>,
]
`;

exports[`userList should render default user list 1`] = `
Array [
  <p>
    Brianna Forbes
  </p>,
  <p>
    Darcy Candice Ball
  </p>,
  <p>
    Hillary Gibbs
  </p>,
  <p>
    Elva Becky Hammond
  </p>,
  <p>
    Enid Rose
  </p>,
  <p>
    Esmeralda Carrillo
  </p>,
]
`;

exports[`userList should render desc user list 1`] = `
Array [
  <p>
    Hillary Gibbs
  </p>,
  <p>
    Esmeralda Carrillo
  </p>,
  <p>
    Enid Rose
  </p>,
  <p>
    Elva Becky Hammond
  </p>,
  <p>
    Darcy Candice Ball
  </p>,
  <p>
    Brianna Forbes
  </p>,
]
`;
